### PR TITLE
Update fake master job

### PR DESCRIPTION
**Pull request by kargakis**
I need to make this optional in the Jenkinsfile so the test job will be able to either test that fake-master can be rebased (merge-and-build.sh) or sync fake-master with enterprise-3.5 (update-fake-master.sh).

@stevekuznetsov @jupierce @tdawson @bbguimaraes

### CODE REVIEW

File: scripts/update-fake-master.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+echo "GOPATH: ${GOPATH}"
+echo "BUILDPATH: ${BUILDPATH}"
+echo "WORKPATH ${WORKPATH}"
+
+# Ensure ssh-agent is running
+eval "$(ssh-agent -s)"
+
+# Add back deploy key for cloning/pushing openshift/ose
+ssh-add -D
+ssh-add ${HOME}/.ssh/id_rsa
```
Comment (bbguimaraes) [inline scripts/update-fake-master.sh]: This is the default location for the key, I don't think we need ssh-agent.

File: scripts/update-fake-master.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+git clone git@github.com:openshift/ose.git
+pushd ose
+git remote add upstream git@github.com:openshift/origin
+git fetch --all
+
+# Check if there is anything to update.
+LATEST_TAG=$(git describe --abbrev=0 --tags)
+LATEST_TITO_COMMIT_MERGE=$(git log -1 $LATEST_TAG --pretty=%s)
+git checkout fake-master
+LATEST_TITO_COMMIT_REBASE=$(git log -1 HEAD --pretty=%s)
```
Comment (bbguimaraes) [inline scripts/update-fake-master.sh]: No need to checkout:

    LATEST_TITO_COMMIT_MERGE=$(git log -1 origin/fake-master --pretty=%s)

File: scripts/update-fake-master.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+if [[ $LATEST_TITO_COMMIT_MERGE == $LATEST_TITO_COMMIT_REBASE ]]; then
+	echo "Nothing to update!"
+	return 0
+fi
+
+# Prepare Origin branch from which we will rebase on top
+git checkout upstream/release-1.5
+git checkout -b rebase-target
+# Start from the last commit that got pulled in OSE during the previous rebase.
+git reset --hard $(git merge-base enterprise-3.5 upstream/release-1.5)
```
Comment (bbguimaraes) [inline scripts/update-fake-master.sh]: This should be enough:

    git branch rebase-target \
        "$(git merge-base enterprise-3.5 upstream/release-1.5)"

File: scripts/update-fake-master.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+# Stage the enterprise branch and work from the new branch to update fake-master
+git checkout enterprise-3.5
+git checkout -b update-fake-master
```
Comment (bbguimaraes) [inline scripts/update-fake-master.sh]: `git checkout enterprise-3.5 -b update-fake-master`

File: scripts/update-fake-master.sh [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+# Reconstruct fake-master carries.
+git cherry-pick $TITO_SQUASH
+VC_COMMIT="$(GIT_REF=enterprise-3.5 hack/vendor-console.sh 2>/dev/null | grep "Vendoring origin-web-console" | awk '{print $4}')"
+git add pkg/assets/bindata.go
+git add pkg/assets/java/bindata.go
+set +e
+# This fails only when there is nothing new to commit, which is normal.
+git commit -m "[DROP] bump origin-web-console ${VC_COMMIT}"
+set -e
```
Comment (bbguimaraes) [inline scripts/update-fake-master.sh]: Surrounding it with `if [ "$(git status --porcelain)" ]; then …; fi` might be
better than disabling error checking altogether.

**Comment by bbguimaraes** [comment]
@kargakis: mostly minor things, overall looks good.

File: scripts/update-fake-master.sh (hunk 2 above)
Comment (kargakis) [inline scripts/update-fake-master.sh]: You mean `LATEST_TITO_COMMIT_REBASE`.

File: scripts/update-fake-master.sh (hunk 5 above)
Comment (kargakis) [inline scripts/update-fake-master.sh]: Agreed, just cp from another branch. Will update

**Comment by kargakis** [comment]
@bbguimaraes thanks, tested locally with the new changes, works fine.

File: scripts/update-fake-master.sh [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+git fetch --all
+
+# Check if there is anything to update.
+# Tags are global to the git repo but can only be accessed from the branches they were tagged in
+# Switch to the enterprise-3.5 branch where our latest tags exist for now.
+git checkout enterprise-3.5
+LATEST_TAG=$(git describe --abbrev=0 --tags)
+LATEST_TITO_COMMIT_MERGE=$(git log -1 $LATEST_TAG --pretty=%s)
+LATEST_TITO_COMMIT_REBASE=$(git log -1 origin/fake-master --pretty=%s)
+if [[ $LATEST_TITO_COMMIT_MERGE == $LATEST_TITO_COMMIT_REBASE ]]; then
```
Comment (stevekuznetsov) [inline scripts/update-fake-master.sh]: I think we switched this in a version of this script in some other branch ( @smunilla ) but you can ask `git` to do this directly for you:

```sh
if git describe --abbrev=0 --tags --exact-match HEAD >/dev/null 2>&1; then
    # latest tag matches HEAD
```

**Pull request merged**